### PR TITLE
made vscode-jupyter to latest LSP

### PR DESCRIPTION
**Pull request by heejaechang**
vscode-jupyter was using old LSP. this will move them to the lastest.

### CODE REVIEW

File: src/intellisense/languageServer.node.ts [hunk 1]
```diff
@@ -79,6 +91,8 @@ class NerfedExecuteCommandFeature implements DynamicFeature<ExecuteCommandRegist
     }
 }
 
+let languageClient: LanguageClient | undefined;
```
Comment (rchiodo) [inline src/intellisense/languageServer.node.ts:94]: This can't be a global variable. The LanguageServer is not a singleton.

**Review by rchiodo — CHANGES_REQUESTED** [review]
I think the language client still needs to be passed into the 'languageServer' class. There's one per notebook interpreter.

File: src/intellisense/languageServer.node.ts (hunk 1 above)
Comment (heejaechang) [reply src/intellisense/languageServer.node.ts:94]: ah.. okay.

File: src/intellisense/languageServer.node.ts [hunk 2]
```diff
@@ -166,33 +199,24 @@ export class LanguageServer implements Disposable {
...
+            await client.start();
 
-            // After starting, wait for it to be ready
-            while (languageClient && !languageClient.initializeResult) {
-                await sleep(100);
-            }
-            if (languageClient) {
-                await languageClient.onReady();
-            }
+            // Expose client once it is fully initialized.
```
Comment (rchiodo) [inline src/intellisense/languageServer.node.ts:216]: I don't think you need this line anymore. Not used.

Comment (heejaechang) [reply src/intellisense/languageServer.node.ts:216]: I think languageClient is still exposed through middleware?

Comment (rchiodo) [reply src/intellisense/languageServer.node.ts:216]: No this variable isn't used. It's just assigned. You use 'client' instead everywhere, so you don't really need this variable.

**Comment by rchiodo** [comment]
Looks like all of our intellisense tests are failing, which is likely caused by this change. 

https://github.com/microsoft/vscode-jupyter/runs/6801323085?check_suite_focus=true

Did you try running a notebook to see if you can get completions still?

**Comment by heejaechang** [comment]
oops. clicked wrong button ..

File: src/intellisense/languageServer.node.ts (hunk 2 above)
Comment (rchiodo) [reply src/intellisense/languageServer.node.ts:215]: Oh didn't notice that as it wasn't in the changes. Yeah looks like it's needed for the lambda capture.

**Comment by rchiodo** [comment]
Looks like the completion tests are still failing. Do you want any help debugging them?

**Review by rchiodo — APPROVED** [review]

**Pull request merged**